packages/dev/loaders/src/USD/usdFileLoader.ts

```diff
@@ -0,0 +1,242 @@
+import type { ISceneLoaderPluginAsync, ISceneLoaderPluginFactory, ISceneLoaderAsyncResult, ISceneLoaderProgressEvent } from "core/Loading/sceneLoader";
+import { registerSceneLoaderPlugin } from "core/Loading/sceneLoader";
+import { USDFileLoaderMetadata } from "./usdFileLoader.metadata";
+import { AssetContainer } from "core/assetContainer";
+import type { Scene } from "core/scene";
+import type { Nullable } from "core/types";
+import type { AbstractMesh } from "core/Meshes/abstractMesh";
+import { Mesh } from "core/Meshes/mesh";
+import { VertexData } from "core/Meshes/mesh.vertexData";
+import { RawTexture } from "core/Materials/Textures/rawTexture";
+import { StandardMaterial } from "core/Materials/standardMaterial";
+import { Texture } from "core/Materials/Textures/texture";
+import { Engine } from "core/Engines/engine";
+import { IsWindowObjectExist } from "core/Misc/domManagement";
+import { Tools } from "core/Misc/tools";
+import { Constants } from "core/Engines/constants";
+import type { USDLoadingOptions } from "./usdLoadingOptions";
+
+declare module "core/Loading/sceneLoader" {
+    // eslint-disable-next-line jsdoc/require-jsdoc
+    export interface SceneLoaderPluginOptions {
+        /**
+         * Defines options for the usd loader.
+         */
+        [USDFileLoaderMetadata.name]: Partial<USDLoadingOptions>;
+    }
+}
+
+/**
+ * @experimental
+ * USD(z) file type loader.
+ * This is a babylon scene loader plugin.
+ */
+export class USDFileLoader implements ISceneLoaderPluginAsync, ISceneLoaderPluginFactory {
+    /**
+     * Defines the name of the plugin.
+     */
+    public readonly name = USDFileLoaderMetadata.name;
+
+    private _assetContainer: Nullable<AssetContainer> = null;
+
+    private readonly _loadingOptions: Readonly<USDLoadingOptions>;
+    /**
+     * Defines the extensions the UDS loader is able to load.
+     * force data to come in as an ArrayBuffer
+     */
+    public readonly extensions = USDFileLoaderMetadata.extensions;
+
+    /**
+     * Creates loader for USD files
+     * @param loadingOptions options for loading and parsing usdz files.
+     */
+    constructor(loadingOptions: Partial<Readonly<USDLoadingOptions>> = USDFileLoader._DefaultLoadingOptions) {
+        this._loadingOptions = loadingOptions;
+    }
+
+    /** @internal */
+    createPlugin(): ISceneLoaderPluginAsync {
+        return new USDFileLoader();
+    }
+
+    private static readonly _DefaultLoadingOptions = {
+        usdLoaderUrl: "https://lighttransport.github.io/tinyusdz/",

```

Comment:
default should be https://cdn.babylonjs.com (it is important so we can use the ability to change base URL)